src/handlers/strategies/directoryListing.ts

```diff
@@ -97,6 +104,49 @@ export function renderDirectoryListing(
   });
 }
 
+/**
+ * Send a request to R2 to get the objects & paths in a directory
+ * @param client {@link S3Client} to use for the request
+ * @param bucketPath Path in R2 bucket
+ * @param cursor Where to begin the request from, for pagination
+ * @param env Worker env
+ * @returns A {@link ListObjectsV2CommandOutput}
+ * @throws When all retries are exhausted and no response was returned
+ */
+async function fetchR2Result(
+  client: S3Client,
+  bucketPath: string,
+  cursor: string | undefined,
+  env: Env
+): Promise<ListObjectsV2CommandOutput> {
+  let retriesRemaining = S3_RETRY_LIMIT;
+  while (retriesRemaining > 0) {
+    try {
+      // Send request to R2
+      const result = await client.send(
+        new ListObjectsV2Command({
+          Bucket: env.BUCKET_NAME,
+          Prefix: bucketPath,
+          Delimiter: '/',
+          MaxKeys: S3_MAX_KEYS,
+          ContinuationToken: cursor,
+        })
+      );
+
+      // Request succeeded, no need for any retries
+      return result;
+    } catch (err) {
+      // Got an error, let's log it and retry
+      console.error(`R2 ListObjectsV2 error: ${err}`);
```

Comment:
Do we use `console.error` anywhere else? Aren't these suppressed?

Reply:
They're still viewable when tailing the worker